amundsen_application/static/js/components/SearchPage/tests/index.spec.tsx

```diff
@@ -273,13 +270,47 @@ describe('SearchPage', () => {
       wrapper.instance().componentDidUpdate(mockPrevProps);
     });
 
-    it('calls setState', () => {
-      expect(setStateSpy).toHaveBeenCalledWith({ selectedTab: ResourceType.table });
-    });
+    describe('called with a new search term', () => {
+      it('calls setState', () => {
+        expect(setStateSpy).toHaveBeenCalledWith({ selectedTab: ResourceType.table });
+      });
 
-    it('calls searchAll', () => {
-      expect(searchAllSpy).toHaveBeenCalledWith(mockSanitizedUrlParams.term, mockSearchOptions);
+      it('calls searchAll', () => {
+        expect(searchAllSpy).toHaveBeenCalledWith(mockSanitizedUrlParams.term, mockSearchOptions);
+      });
     });
+
+    describe('called with a new page but with the same search term', () => {  
```

Comment:
We can condense this by leveraging the setup that was used initially for these tests. This is because this test case doesn't necessarily need a new setup, we only have to call `wrapper.instance().componentDidUpdate()` with some new `mockPrevProps` and verify what happens with the `searchAllSpy`.
```
it('does not call searchAll if called with the same search term', () => {
    /* assuming we use the same setup from the previous tests and don't have to write all the boiler plate again */
    searchAllSpy.mockClear(); // because another test will have called it.
     const mockPrevProps = {
          location: {
            search: '/search?searchTerm=current&current=table&pageIndex=1', 
            pathname: 'mockstr',
            state: jest.fn(),
            hash: 'mockstr',
          }
     };
    wrapper.instance().componentDidUpdate(mockPrevProps);
    expect(searchAllSpy).not.toHaveBeenCalled();
});
```

The above should work, but let me know if you run into any issues.